File: python_collatz/collatz.py
import time


def collatz(seed: int) -> int:
    steps = 0
    while seed > 1:
        if seed % 2 == 0:
            seed /= 2
        else:
            seed = (3 * seed) + 1
        steps += 1
    
    return steps


def main():
    start = time.time()
    for seed in range(1, 1_000_000_001):
        steps = collatz(seed)
        if (seed % 1_000_000 == 0):
            print(f"Seed {seed} Steps: {steps}")
    
    stop = time.time()
    elapsed = stop - start
    print(f"Took {elapsed} seconds")


if __name__ == '__main__':
    main()

File: c_collatz/collatz.c
#include <stdio.h>
#include <time.h>

long collatz(long seed) {
    long steps = 0;
    long n = seed;
    while (n > 1) {
        if (n % 2 == 0) {
            n /= 2;
        } else {
            n = (n * 3) + 1;
        }
        steps++;
    }
    return steps;
}

int main() {
    clock_t start = clock();
    for (long seed = 1; seed <= 1000000000L; seed++) {
        long collatz_result = collatz(seed);
        if (seed % 1000000L == 0) {
            printf("Seed: %ld Steps: %ld\n", seed, collatz_result);
        }
    }
    clock_t end = clock();
    double elapsed = (double)(end - start) / CLOCKS_PER_SEC;
    printf("Took %f seconds", elapsed);
    return 0;
}